src/Sparrow/Json/ArenaMemoryAllocator.cs

```diff
@@ -308,11 +309,18 @@ public override string ToString()
         }
 
         public void Dispose()
+        {
+            Dispose(true);
+        }
+        public void Dispose(bool disposing)
```

Comment:
Probably not required. The only thing you do differently is the lock on the `this`, and we _know_ that can't happen here.

Reply:
what can't happen there? the locking of this?

Reply:
Yes, if we are in the finalizer, we know no one is locking on `this`